providers/amazon/src/airflow/providers/amazon/aws/executors/aws_lambda/lambda_executor.py

```diff
@@ -0,0 +1,488 @@
+
+        For lambda: Use get_function to test if the lambda connection works and the function can be
+        described.
+        For SQS: Use get_queue_attributes is used as a close analog to describe to test if the SQS
+        connection is working.
+        """
+        self.IS_BOTO_CONNECTION_HEALTHY = False
+
+        try:
+            self.log.info("Checking Lambda and SQS connections")
+            # Check Lambda health
+            lambda_get_response = self.lambda_client.get_function(FunctionName=self.lambda_function_name)
+            if self.lambda_function_name not in lambda_get_response["Configuration"]["FunctionName"]:
+                raise AirflowException("Lambda function %s not found.", self.lambda_function_name)
+            self.log.info(
+                "Lambda connection is healthy and function %s is present.", self.lambda_function_name
+            )
+
+            def _check_queue(queue_url):
+                sqs_get_queue_attrs_response = self.sqs_client.get_queue_attributes(
+                    QueueUrl=queue_url, AttributeNames=["ApproximateNumberOfMessages"]
+                )
+                try:
+                    approx_num_msgs = sqs_get_queue_attrs_response.get("Attributes").get(
+                        "ApproximateNumberOfMessages"
+                    )
+                    self.log.info(
+                        "SQS connection is healthy and queue %s is present with %s messages.",
+                        queue_url,
+                        approx_num_msgs,
+                    )
+                except Exception:
+                    raise AirflowException("SQS queue %s not found.", queue_url)
+
+            # Check SQS results queue
+            _check_queue(self.sqs_queue_url)
+            # Check SQS dead letter queue
+            _check_queue(self.dlq_url)
+
+            # If we reach this point, both connections are healthy and all resources are present
+            self.IS_BOTO_CONNECTION_HEALTHY = True
+        except Exception:
+            self.log.exception("Lambda Executor health check failed")
+            raise AirflowException(
+                "The Lambda executor will not be able to run Airflow tasks until the issue is addressed."
+            )
+
+    def load_connections(self, check_connection: bool = True):
+        """
+        Retrieve the AWS connection via Hooks to leverage the Airflow connection system.
+
+        :param check_connection: If True, check the health of the connection after loading it.
+        """
+        self.log.info("Loading Connections")
+        aws_conn_id = conf.get(CONFIG_GROUP_NAME, AllLambdaConfigKeys.AWS_CONN_ID, fallback="aws_default")
+        region_name = conf.get(CONFIG_GROUP_NAME, AllLambdaConfigKeys.REGION_NAME, fallback=None)
+        self.sqs_client = SqsHook(aws_conn_id=aws_conn_id, region_name=region_name).conn
+        self.lambda_client = LambdaHook(aws_conn_id=aws_conn_id, region_name=region_name).conn
```

Comment:
Do not you think we should cache them?

Reply:
What do you mean? The same connection is used unless we detect an issue with it (like expired tokens) and then this function will be called again only in that case.

Reply:
I meant the hooks, do you think it is worth caching them with `@cached_property`. As far as I know, hook creation can be heavy?

Reply:
We actually need new hooks to get new connections (since the hooks cache their connections), which is the goal here to get a brand new connection since the last one is unhealthy. And again, we only call it when the connection is unhealthy which should be rare, and if you have an unhealthy connection your executor will be completely broken and and non-functioning, so I'm sure the user will be happy to pay a small time/cpu price to fix up everything!

Reply:
Indeed, sounds very valid to me :) Thanks for clarifying